docs/src/test-configuration.md

```diff
@@ -149,6 +114,358 @@ test.describe('headed block', () => {
+
+- `baseURL` - Base URL used for all pages in the context. Allows navigating by using just the path, for example `page.goto('/settings')`.
+- `browserName` - Name of the browser that will run the tests, one of `chromium`, `firefox`, `webkit`.
+- `bypassCSP` - Toggles bypassing page's Content-Security-Policy. Useful when CSP includes production origin.
+- `channel` - Browser channel to use. [Learn more](./browsers.md) about different browsers and channels.
+- `headless` - Whether to run browser in headless mode.
+- `viewport` - Viewport used for all pages in the context.
+- `storageState` - Populates context with given storage state. Useful for easy authentication, [learn more](./auth.md).
+
+```js js-flavor=js
+// @ts-check
+
+/** @type {import('@playwright/test').PlaywrightTestConfig} */
+const config = {
+  use: {
+    baseURL: 'http://localhost:3000',
+    browserName: 'firefox',
+    headless: true,
+  },
+};
+
+module.exports = config;
+```
+
+```js js-flavor=ts
+import { PlaywrightTestConfig } from '@playwright/test';
+const config: PlaywrightTestConfig = {
+  use: {
+    baseURL: 'http://localhost:3000',
+    browserName: 'firefox',
+    headless: true,
+  },
+};
+export default config;
+```
+
+## Emulation
+
+Playwright can [emulate different environments](./emulation.md) like mobile device, locale or timezone.
+
+Here is an example configuration that runs tests in "Pixel 4" and "iPhone 11" emulation modes. Note that it uses the [projects](./test-advanced.md#projects) feature to run the same set of tests in multiple configurations.
+
+```js js-flavor=js
+// playwright.config.js
+// @ts-check
+const { devices } = require('playwright');
+
+/** @type {import('@playwright/test').PlaywrightTestConfig} */
+const config = {
+  projects: [
+    // "Pixel 4" tests use Chromium browser.
+    {
+      name: 'Pixel 4',
+      use: {
+        browserName: 'chromium',
+        ...devices['Pixel 4'],
+      },
+    },
+
+    // "iPhone 11" tests use WebKit browser.
+    {
+      name: 'iPhone 11',
+      use: {
+        browserName: 'webkit',
+        ...devices['iPhone 11'],
+      },
+    },
+  ],
+};
+
+module.exports = config;
+```
+
+```js js-flavor=ts
+// playwright.config.ts
+import { PlaywrightTestConfig } from '@playwright/test';
+import { devices } from 'playwright';
+
+const config: PlaywrightTestConfig = {
+  projects: [
+    // "Pixel 4" tests use Chromium browser.
+    {
+      name: 'Pixel 4',
+      use: {
+        browserName: 'chromium',
+        ...devices['Pixel 4'],
+      },
+    },
+
+    // "iPhone 11" tests use WebKit browser.
+    {
+      name: 'iPhone 11',
+      use: {
+        browserName: 'webkit',
+        ...devices['iPhone 11'],
+      },
+    },
+  ],
+};
+export default config;
+```
+
+You can also specify options separately, instead of using predefined `devices`, or configure locale, geolocation, timezone and more:
```

Comment:
```suggestion
You specify options separately instead of using predefined devices. There are also more options such as locale, geolocation, and timezone which can be configured.
```